.travis.yml

```diff
@@ -11,22 +11,25 @@ install:
 before_script:
     - "export DISPLAY=:99.0"
     - "Xvfb :99 -screen 0 1920x1080x24+32 +extension GLX +extension RANDR > /dev/null 2>&1 &"
-    - sleep 3 # give xvfb some time to start
 
 with_content_shell: true
 
 # Job sections
 jobs:
   include:
     - stage: Lint
-      before_install: skip
       before_script: skip
       install: pip install tox
       script: tox -e flake8
-    - stage: Ubuntu 16.04 Image search and mouse movement tests
+    - stage: Build
       os: linux
       dist: xenial
       script: pipenv run iris -h
+    - stage: Smoke Tests
+      install: pip install tox
+      script:
+        - tox -e default_config
+        - tox -e custom_config
 
```

Comment:
What I see here are unit tests, written against code that is run in a context outside of the project. I think we should rename this stage to "Unit Tests" and move it before the "Build" stage, as it is more low-level and errors here should prevent us from going any further in the validation process. Do you agree?

Also, the "Build" stage could be followed by a "Smoke Tests" stage that invokes a script that actually performs integration scenarios. Much like how the "Build" stage currently just invokes the -h flag, a "Smoke Tests" stage could run commands such as `pipenv run iris sample -t specific_test <args>` and confirm specific behavior. However, we can cover that later.

Reply:
I really wouldn't consider these unit tests. The only piece of iris core code it touches is the ability to set options. They do a low level check to ensure the tests will be able to run as expected.

I see your point about the order of the stages. the Build stage takes a while so if Smoke will fail we should fail faster. I've just pushed those changes to my fork and the travis builds are passing.

I could add more smoke tests for sure, using the `sample` target

Reply:
[Unit test](https://en.wikipedia.org/wiki/Unit_testing) vs [smoke test.](http://softwaretestingfundamentals.com/smoke-testing/)

I think the tests you've written fit the description of unit tests, but do not verify a successful build, which is why they shouldn't be considered smoke tests either. I think they are useful, and if we'd like to write other low-level tests like this, let's run them together before the "Build" step.

We have to build before we can run smoke tests, if we are to follow the definition.